documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3030,4 +3032,86 @@ The following would be valid:
   }
 ...
 ```
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+
+### <a name="r4029" ></a>R4029 UniqueClientParameterName
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  Do not have duplicate name of client parameter name, make sure every client parameter name unique.
+
+**Description** : Client parameter name must be unique. Duplicate client parameter name cause SDK conflict. 
+
+**Why this rule is important**: Make sure no conflict in SDK generation.
+
```

Comment:
Seems duplicate with the Description, you can provide the reason  from dapeng : 
This may cause a problem when different swagger files come together. If two APIs with different client name have the same client parameter `subscriptionId`, but with different reference name in swaggers, the generated model will also have two clients with two client parameters `subscriptionId` and `subscriptionId1` (the latter one has been renamed to avoid collision).
We should ensure that the client parameters are all unique.

Reply:
Good point.